Support HOS16.0.0

zdm65477730 opened the pull request:
1. Support HOS16.0.0.
2. Fix play time in total time exceed 100%.
3. Add year suffix for some local lang, e.g. Chinese.
4. Allow run under Applet mode.
5. Make sure all account play events in list.
6. Optimize the memory usage.
7. Fix ghost record issue in all views(All, Recent and Hidden view).
8. Fix crash when deleting import data.
9. Correct the time unit(time_t in second should be uint64_t).

zdm65477730 committed:
support hos16.0.0 and min changes

1. Support HOS16.0.0.
2. Fix play tiem in total time exceed 100%.
3. Add year suffix.
4. Allow run under Applet mode.
5. Make sure all account play events in list.

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update NX.cpp

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Optimize the memory usage

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
use the correct time unit

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
solve the possiable ghost record issue

There are some ghost records in the export json file
"id": "0000000000000000",
"name": "",
"summary": {
    "firstPlayed": 0,
    "lastPlayed": 0,
    "launches": 0,
    "playtime": 0
}

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix ghost record in hidden view and fix crash when deleting importing data

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
bump version to 1.5.0

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
make sure the title vector has no zero id

Signed-off-by: Damien Zhao <zdm65477730@126.com>

BullishVince commented:
Can you please merge this into master and make it as a release? @tallbl0nde

haren724 commented:
Hi, @zdm65477730. Is your git repository still being stored locally? The submodule 'SimpleIniParser' doesn't exist anymore. I planned to add a tiny functionality that play an .wav audio file (whistle twice sound in stock) after launching this app from the user page. Could you send me a copy or any other valid link of this submodule.

Thanks!

ghost commented:
I found simpleiniparser that are active here
https://github.com/zdm65477730/SimpleIniParser
and here
https://github.com/HamletDuFromage/Simple-INI-Parser

A tried too some changes but when i enter the app it crash with error. But compiles without problem. Could you please test it with your devkitpro if I set it correctly?

https://www.mediafire.com/file/gf9mqvak8oinjuo/NX-Activity-Log-master-Test.zip/file

Thank you

haren724 commented:
> I found simpleiniparser that are active here https://github.com/zdm65477730/SimpleIniParser and here https://github.com/HamletDuFromage/Simple-INI-Parser
> 
> A tried too some changes but when i enter the app it crash with error. But compiles without problem. Could you please test it with your devkitpro if I set it correctly?
> 
> https://www.mediafire.com/file/gf9mqvak8oinjuo/NX-Activity-Log-master-Test.zip/file
> 
> Thank you

@Lukas2288
I got the same result. It causes a kernel panic in applet mode and a exefs crash in exefs mode.
Trying to figure it out.

ghost commented:
> > I found simpleiniparser that are active here https://github.com/zdm65477730/SimpleIniParser and here https://github.com/HamletDuFromage/Simple-INI-Parser
> > A tried too some changes but when i enter the app it crash with error. But compiles without problem. Could you please test it with your devkitpro if I set it correctly?
> > https://www.mediafire.com/file/gf9mqvak8oinjuo/NX-Activity-Log-master-Test.zip/file
> > Thank you
> 
> @Lukas2288 I got the same result. It causes a kernel panic in applet mode and a exefs crash in exefs mode. Trying to figure it out.

I think that there should be problem with new version of devkit libraries. I´ve got version 1.4 which i updated with my theme and adjustments and it compiles and runs fine. But 2 years later i cannot start it anymore, when i recompile it.

zdm65477730 committed:
uplift submodues

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update for new libnx

Signed-off-by: Damien Zhao <zdm65477730@126.com>

AlM2025 commented:
Please add support to HOS19.0.0.

zdm65477730 committed:
updatge submoudle

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update lib

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update Makefile and system json

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update Makefile for new libnx

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update the update checking source

Signed-off-by: Damien Zhao <zdm65477730@126.com>

yyoossk commented:
Added Japanese language. Please merge, build, and release!

zdm65477730 committed:
Create main.yml

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
Update Makefile

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
pull jpn&kr lang commits

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
add Japanese support

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
uplift Aether

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
uplift Aether

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Optimize the loading performance when dealing with a large number of data entries

- Fixed APP stuck on black screen when there are a large number of game
  data entries
- Optimized the loading performance of the game data enteries.
- Add debug traces.
- Bump to v1.5.2

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix total play time in recent activity view mismatch

Bump version to v1.5.3

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix total play time in recent activity view mismatch

Revert the pervious commit and update the fix.

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Make the recorded time more accurate

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update for use the same API for fetch user's game play data

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix data missing in year view

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
some fixes

1. Fixed crash when deleting import json file.
2. Updates all lang json files.
3. Fixed first/last play data missing for imported play data.

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fixed big when creating messageboxes and updated for full translations

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
bump to v1.5.4

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix bug that miss some play data

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
min update

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
min fix

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Bump to v1.5.5

1. Fix some play data missing in All&Recently View.
2. Fix debug log write interface.

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
add Ukrainian lang support

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Bump to v1.5.6

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
make sure TitleID 0 won't be added

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix titleID 0 be added by mistake

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Update README.md

zdm65477730 committed:
fix the time count loss due to DST

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
bump to v1.5.7

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix wrong steady time in countPlaytime()

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
min fix

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Revert "min fix"

This reverts commit aa8db778e18519ca8cd1ce8f760c9d9aded192c3.

zdm65477730 committed:
Revert "fix wrong steady time in countPlaytime()"

This reverts commit fc5facf2b58292725af644c7e2d69359f182d401.

zdm65477730 committed:
Revert "fix the time count loss due to DST"

This reverts commit c8c77d52c1ebc1c12eb2a5c5b4e4288d6011a251.

zdm65477730 committed:
fix 1 hour missing in Month view

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
fix 1 hour missing in Recent Activity view

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update main.yaml

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Update main.yml

zdm65477730 committed:
Revert "Update main.yml"

This reverts commit 62b7975414b56e9de4d7b7abb0e3042aedaafe65.

zdm65477730 committed:
Revert "update main.yaml"

This reverts commit 4c8826c54141b58d9d7193c05be3ac488266a0a6.

zdm65477730 committed:
Optimize gameplay data loading and UI creation logic

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
min update

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
min update for thread memory

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
min update

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
min fix

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Revert "min fix"

This reverts commit 9cd30846c3bc9ee35369ae800bfa5871443f862b.

zdm65477730 committed:
Revert "min update for thread memory"

This reverts commit 1208f83d83e504452c45042bbabe0bca74d3ec7a.

zdm65477730 committed:
Revert "min update"

This reverts commit 656148d52790ad8b3e6411dad0c9f73e7119ee90.

zdm65477730 committed:
Revert "min update"

This reverts commit 689bdad1ab5d72234fb74b7a5dff7e26873a0ec3.

zdm65477730 committed:
Revert "Optimize gameplay data loading and UI creation logic"

This reverts commit aa87410934634e2c7428c14a3260d555df14b4b2.

zdm65477730 committed:
revert libs

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
update Makefile

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Optimize gameplay data loading and UI creation logic

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Bump to v1.0.8

zdm65477730 committed:
fix issue22

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
uplift lang

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
use origin libnx

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
align with latest nx-hbloader to fix My Page crash

Signed-off-by: Damien Zhao <zdm65477730@126.com>

zdm65477730 committed:
Bump to v1.5.9
